### PR TITLE
use webassembly-interpreter to parse WASM

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
refactor

**Did you add tests for your changes?**
existing tests

**If relevant, link to documentation update:**
n/a

**Summary**
#6530

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "tapable": "^1.0.0-beta.5",
     "uglifyjs-webpack-plugin": "^1.1.1",
     "watchpack": "^1.4.0",
+    "webassembly-interpreter": "^0.0.29",
```
Comment (xtuc) [inline package.json]: It's probably safer if I make a 1.x release now, or doesn't it matter?

Comment (sokra) [reply package.json]: `^` has no effect with `0.0.x` versions. So it's like `"0.0.29"`.

But `1.x` would be good anyway, if you want to push bugfix releases.

Comment (xtuc) [reply package.json]: Yes ok, I would like to nail down a few PR before releasing a major version. I can upgrade it in Webpack then.

File: lib/WebAssemblyParser.js [hunk 2]
```diff
@@ -2,10 +2,9 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-/* globals WebAssembly */
 "use strict";
 
-// Syntax: https://developer.mozilla.org/en/SpiderMonkey/Parser_API
+var Tools = require("webassembly-interpreter/lib/tools");
```
Comment (ooflorent) [inline lib/WebAssemblyParser.js]: Could you use `const` to be consistent with other `require` statements?

**Comment by xtuc** [comment]
Ah yeah, I didn't try with the `order` wasm test suite. I'm going to fix that.

**Comment by sokra** [comment]
@xtuc could you take a look at these errors?

**Comment by xtuc** [comment]
@sokra yes! see https://github.com/webpack/webpack/pull/6545

**Comment by xtuc** [comment]
Prettier caused the conflicts here, you can solve it by running prettier on `lib/WebAssemblyParser.js`.

**Comment by xtuc** [comment]
We can close this btw, I included it in https://github.com/webpack/webpack/pull/6709

**Pull request closed without merging**